freeipa-client/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -704,8 +639,28 @@ public TopologySegment deleteTopologySegment(String topologySuffixCn, TopologySe
         return (TopologySegment) invoke("topologysegment_del", flags, params, TopologySegment.class).getResult();
     }
 
+    public void callBatch(BiConsumer<String, String> warnings, List<Object> operations,
+            Set<FreeIpaErrorCodes> acceptableErrorCodes) throws FreeIpaClientException {
+        callBatch(warnings, operations, DEFAULT_BATCH_CALL_PARTITION_SIZE, acceptableErrorCodes);
+    }
+
+    public void callBatch(BiConsumer<String, String> warnings, List<Object> operations, Integer partitionSize,
+            Set<FreeIpaErrorCodes> acceptableErrorCodes) throws FreeIpaClientException {
+        List<List<Object>> partitions = Lists.partition(operations, partitionSize);
+        for (List<Object> operationsPartition : partitions) {
+            BatchOperation.get(operationsPartition, warnings, acceptableErrorCodes).invoke(this);
+        }
+    }
+
+    public static void checkIfClientStillUsable(FreeIpaClientException e) throws FreeIpaClientException {
```

Comment:
why static?

Reply:
logic depends only on the parameter, not on the freeipaclient object. but it seems, FreeIpaClientExceptionUtil has the same logic, so i'll remove this

Reply:
but I think it's harder to mock, so I'm against using static. Also in this case it wouldn't make any difference.

Reply:
that's a valid point, i'll remove static, and keep this method because it has logging